### PR TITLE
Add WATERMARK clause

### PR DESCRIPTION
**Pull request by emile-00**
## Info

- **Description**

  Add guide on WATERMARK clause. Only added file, have not added to sidebar yet. 

- **Notes**

  [ Include any supplementary context or references here. ]

- **Related code PR**

  [ Provide a link to the relevant code PR here, if applicable. ]

- **Related doc issue**
  
  Resolves [ Provide a link to the relevant doc issue here, if applicable. ]

## For reviewers

- **Preview**

  [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **upcoming** version of the documentation. ]

- **Key points**

  [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

File: docs/sql/sql-watermarks.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+---
+id: sql-watermarks
+slug: /sql-watermarks
+title: Watermarks
+---
+
+In stream processing, watermarks are integral when using event time processing logic with event time based operations. Watermarks are like markers or signals that track the progress of event time, allowing you to process events within their corresponding time windows. A watermark is an estimate of the maximum event time observed so far, or a threshold indicating that events received so far have a timestamp later than or equal to the current watermark. Events that arrive with a timestamp earlier than the current watermark are considered late and are not processed within its time window.
```
Comment (hengm3467) [inline docs/sql/sql-watermarks.md]: It's better to include some examples here.

File: docs/sql/sql-watermarks.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+title: Watermarks
+---
+
+In stream processing, watermarks are integral when using event time processing logic with event time based operations. Watermarks are like markers or signals that track the progress of event time, allowing you to process events within their corresponding time windows. A watermark is an estimate of the maximum event time observed so far, or a threshold indicating that events received so far have a timestamp later than or equal to the current watermark. Events that arrive with a timestamp earlier than the current watermark are considered late and are not processed within its time window. 
+
+## Syntax
+
+Watermarks can be generated directly on sources.
+
+The syntax of the `WATERMARK` clause in RisingWave is as follows:
```
Comment (hengm3467) [inline docs/sql/sql-watermarks.md]: The syntax is used in CREATE SOURCE (and CREATE TABLE WITH CONNECTOR), so we'll also need to update that articles.

File: docs/sql/sql-watermarks.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
+---
+id: sql-watermarks
+slug: /sql-watermarks
+title: Watermarks
+---
+
+In stream processing, watermarks are integral when using event time processing logic with event time based operations. Watermarks are like markers or signals that track the progress of event time, allowing you to process events within their corresponding time windows. A watermark is an estimate of the maximum event time observed so far, or a threshold indicating that events received so far have a timestamp later than or equal to the current watermark. Events that arrive with a timestamp earlier than the current watermark are considered late and are not processed within its time window. 
+
+Let us go over an example on how watermarks are generated and utilized during window computations. Say the following events and their corresponding event-time timestamps arrive.
```
Comment (emile-00) [inline docs/sql/sql-watermarks.md]: @TennyZhuang Please review this explanation when you get the chance. Thank you!

**Review by TennyZhuang — APPROVED** [review]
LGTM

**Pull request merged**